src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOS_LandAssimGridComp.F90

```diff
@@ -2602,28 +2630,48 @@ subroutine get_mwrtm_param(MAPL, clock, N_catl, INTERNAL, rc)
     real, dimension(:), pointer :: BH
     real, dimension(:), pointer :: BV
     real, dimension(:), pointer :: LEWT
-    real, dimension(:), pointer :: DCATAU   ! needed?? declared as global var in module!?  reichle, 27Jun2021
+    
+    real, dimension(:), pointer :: VEGOPACITY
     
     integer :: N_catl_tmp, n, mpierr, status
     logical :: mwp_nodata, all_nodata_l
 
-    character(len=ESMF_MAXSTR)  :: TAUFile
+    character(len=ESMF_MAXSTR)  :: VEGOPACITYFile
     type(ESMF_Time)             :: CURRENT_TIME
- 
+
+    ! it looks like this if statement prevents us from reading the vegetation opacity from the file after
+    ! the first call to this subroutine
+    ! we don't want to re-do everything in this subroutine at every 3-hour analysis step, so we'll need
+    ! to find a way to keep reading the VEGOPACITY file and replacing the mwrtm_param%vegopacity field with
+    ! time-varying data in some other way.
+    ! - reichle 13 July 2021
     if(allocated(mwRTM_param)) then
        _RETURN(_SUCCESS)
     endif
 
-    allocate(DCATAU(N_catl))
-    call MAPL_GetResource(MAPL, TAUFile, label = 'TAU_FILE:', &
-        default = 'tau.dat', RC=STATUS )
+    allocate(VEGOPACITY(N_catl))
```

Comment:
```suggestion
    allocate(VEGOPACITY(N_catl), source=MAPL_UNDEF)
```

Reply:
it will be assigned to mwRTM structure if there is no file. So the default value here is necessary.